chef/automate
# Nodemanager deleting nodes for ingest rpc delete requests

### :nut_and_bolt: Description: What code changed, and why?

When the ingest-service's "/ingest/events/chef/nodedelete" or "/ingest/events/chef/node-multiple-deletes" endpoints are called the nodes are also deleted from the nodemanager.

### :chains: Related Resources
https://github.com/chef/automate/issues/616

### :athletic_shoe: How to Build and Test the Change
1. `build components/ingest-service && start_all_services`
1. Add an old CCR with` send_chef_run_example`
1. Run the below command to see the node in the nodemanager. 
```
curl -f --insecure -H "api-token: $(get_admin_token)" https://localhost/api/v0/nodes/search -d '{
    "filters": [
      {"key": "manager_id", "values": [""]},
      {"key": "name", "values": ["insights.B"]}
    ]
  }' | jq
```
1. Go to the Client Runs page and get the node's ID
1. Add the node's ID to the below command to delete the node. 
```
curl -f --insecure -H "api-token: $(get_admin_token)" https://localhost/api/v0/ingest/events/chef/node-multiple-deletes -d '{ "node_ids": ["<node-id>"] }'
```
1. Run the below command to ensure the node is deleted.  
```
curl -f --insecure -H "api-token: $(get_admin_token)" https://localhost/api/v0/nodes/search -d '{
    "filters": [
      {"key": "manager_id", "values": [""]},
      {"key": "name", "values": ["insights.B"]}
    ]
  }' | jq
```

### :white_check_mark: Checklist
